c7n/resources/s3.py

```diff
@@ -2001,7 +2001,13 @@ class SetInventory(BucketActionBase):
 
     def process(self, buckets):
         with self.executor_factory(max_workers=2) as w:
-            list(w.map(self.process_bucket, buckets))
+            futures = {w.submit(self.process_bucket, bucket): bucket for bucket in buckets}
+            for future in as_completed(futures):
+                bucket = futures[future]
+                try:
+                    future.result()
+                except Exception as e:
+                    self.log.error('Message: %s Bucket: %s', e, bucket['Name'])
```

Comment:
thanks, much nicer, on the log error would be worth noting the action being attempted, message is bit generic.

Reply:
The exception has the full text of the error, including the action

Reply:
cool